### PR TITLE
Improve OBD data screen; removed used devices from found bounded devi…

### PR DESCRIPTION
**Pull request by Corwin-Kh**
…ces list

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/plugins/odb/dialogs/OBDDevicesSearchFragment.kt [hunk 1]
```diff
@@ -107,9 +107,14 @@ class OBDDevicesSearchFragment : OBDDevicesBaseFragment(),
 
 	override fun onStart() {
 		super.onStart()
-		pairedDevicesAdapter.items =
-			vehicleMetricsPlugin?.getPairedOBDDevicesList(requireActivity()) ?: emptyList()
-		updateCurrentStateView()
+		vehicleMetricsPlugin?.let { plugin ->
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/plugins/odb/dialogs/OBDDevicesSearchFragment.kt:110]: We could get vehicleMetricsPlugin via requirePlugin & remove "?"

File: OsmAnd/src/net/osmand/plus/plugins/odb/dialogs/OBDMainFragment.kt [hunk 2]
```diff
@@ -27,10 +27,29 @@ import net.osmand.util.Algorithms
...
+	}
+
+	data class OBDDataItem(val dataType: OBDDataType, val widget: OBDComputerWidget)
+
 	private val handler = Handler(Looper.getMainLooper())
 	private val items = mutableListOf<Any>()
 
-	private lateinit var adapter: OBDMainFragmentAdapter
+	private var adapter: OBDMainFragmentAdapter? = null
 	private var progress: View? = null
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/plugins/odb/dialogs/OBDMainFragment.kt:53]: Why we make it nullable?

Comment (Corwin-Kh) [reply OsmAnd/src/net/osmand/plus/plugins/odb/dialogs/OBDMainFragment.kt:53]: updateNightMode() could be called before adapter is initialized

File: OsmAnd/src/net/osmand/plus/plugins/odb/dialogs/OBDMainFragment.kt [hunk 3]
```diff
@@ -301,11 +321,17 @@ class OBDMainFragment : OBDDevicesBaseFragment(), VehicleMetricsPlugin.Connectio
...
 
 	override fun onForgetSensorConfirmed(deviceId: String) {
 		vehicleMetricsPlugin?.removeDeviceToUsedOBDDevicesList(deviceId)
 		view?.let { setupUI(it) }
 	}
+
+	override fun updateNightMode() {
+		super.updateNightMode()
+		adapter?.nightMode = nightMode
+		adapter?.notifyDataSetChanged()
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/plugins/odb/dialogs/OBDMainFragment.kt:335]: It looks somewhat strange, correct nightMode should be set in onCreateView, also theme change should force layout recreation after this [method](https://github.com/osmandapp/Osmand/blob/fecbd55b6d6cd4b30a232b8ab0e1b2ba2bceadf8/OsmAnd/src/net/osmand/plus/helpers/MapFragmentsHelper.java#L117)

**Pull request merged**